src/v0/destinations/salesforce/transform.js

```diff
@@ -212,21 +227,31 @@ async function getSalesforceIdFromPayload(message, authorizationData, destinatio
       throw new InstrumentationError('Invalid Email address for Lead Objet');
     }
     const leadQueryUrl = `${authorizationData.instanceUrl}/services/data/v${SF_API_VERSION}/parameterizedSearch/?q=${email}&sobject=Lead&Lead.fields=id,IsConverted,ConvertedContactId,IsDeleted`;
+    const finalHeader =
+      authorizationFlow === OAUTH
+        ? { Authorization: `Bearer ${authorizationData.token}` }
+        : { Authorization: authorizationData.token };
+
     // request configuration will be conditional
     const { processedResponse: processedLeadQueryResponse } = await handleHttpRequest(
       'get',
       leadQueryUrl,
       {
-        headers: { Authorization: authorizationData.token },
+        headers: finalHeader,
```

Comment:
Can't we use getAuthHeader({ authorizationFlow, authorizationData }) here ?

Reply:
done